From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Wed, 18 Mar 2020 21:53:05 -0500
Subject: [PATCH] Add getPlayerCause to LightningStrike entities

Tridents with the channeling enchantment will create
lightning if thrown during a thunderstorm.
This returns if a player caused the lightning strike.

diff --git a/src/main/java/org/bukkit/entity/LightningStrike.java b/src/main/java/org/bukkit/entity/LightningStrike.java
index 2c81a3f685588431a3c7675c84b35a28975232af..ea82dfb7deab96af9646d24ac9a7d1c4cbe369a6 100644
--- a/src/main/java/org/bukkit/entity/LightningStrike.java
+++ b/src/main/java/org/bukkit/entity/LightningStrike.java
@@ -1,6 +1,7 @@
 package org.bukkit.entity;
 
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable; // EMC
 
 /**
  * Represents an instance of a lightning strike. May or may not do damage.
@@ -65,4 +66,14 @@ public interface LightningStrike extends Entity {
      */
     void setLifeTicks(int lifeTicks);
     // Paper end
+
+    // EMC start
+    /**
+     * Returns the player that caused this ligntning using a trident or null if not applicable.
+     *
+     * @return the player that caused the strike or else null
+     */
+    @Nullable
+    public Player getPlayerCause();
+    // EMC end
 }
